src/extension/decorations/closing_labels_decorations.ts

```diff
@@ -12,15 +13,24 @@ export class ClosingLabelsDecorations implements vs.Disposable {
 	private activeEditor?: vs.TextEditor;
 	private closingLabels?: as.AnalysisClosingLabelsNotification;
 
-	private readonly decorationType = vs.window.createTextEditorDecorationType({
-		after: {
-			color: new vs.ThemeColor("dart.closingLabels"),
-			margin: "2px",
-		},
-		rangeBehavior: vs.DecorationRangeBehavior.ClosedOpen,
-	});
+	private decorationType!: vs.TextEditorDecorationType;
```

Comment:
Do we need these `!`s here? It's not clear to me what they're doing.

Reply:
Without it, since we are updating the property only inside the (current, will rename) `buildDecorationType` we get:

```console
Property 'decorationType' has no initializer and is not definitely assigned in the constructor.
```

The `!` there fixes it. I was getting the same error on the `closingLabelsPrefix` before I initiated it there so this one is not needed.

Reply:
Ah, got it - I didn't realise you could silence that in this way. I guess this is necessary unless we assign it in the constructor, but that makes it harder to share the `buildDecorationType` implementation. Keep it is fine.